### PR TITLE
Sync Plugin Interface

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
As title.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
I need maintainers to take a look at whether this implementation is good enough.
Currently, the "core. go" file is not fully completed, but you can take a look at the interface.

## Tracking Issue
https://github.com/flyteorg/flyteidl/pull/436
https://github.com/flyteorg/flytekit/pull/1822
## Follow-up issue
https://github.com/flyteorg/flyte/issues/3936

### CODE REVIEW

**Comment by Future-Outlier** [comment]
@eapolinario  I have moved the PR to here.
https://github.com/flyteorg/flyte/pull/4107

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4131. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.
